views/audit/bigquery_audit_log_v2.sql

```diff
@@ -0,0 +1,1161 @@
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.routineCreation.jobName'),
+            "/")[SAFE_OFFSET(3)]
+    ) AS jobId,
+  FROM `project_id.dataset_id.cloudaudit_googleapis_com_data_access`
+),
+/*
+ * RoutineChange: Routine change event
+ * https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/BigQueryAuditMetadata#routinechange
+ */
+routineChangeEvent AS (
+  SELECT
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineChange.routine.routineName') AS routineChangeName,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineChange.routine.createTime') AS routineChangeCreateTime,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineChange.routine.updateTime') AS routineChangeUpdateTime,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineChange.reason') AS routineChangeReason,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineChange.jobName') AS routineChangeJobName,
+    CONCAT(
+      SPLIT(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+            '$.routineChange.jobName'),
+            "/")[SAFE_OFFSET(1)],
+      ":",
+      SPLIT(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.routineChange.jobName'),
+            "/")[SAFE_OFFSET(3)]
+    ) AS jobId,
+  FROM `project_id.dataset_id.cloudaudit_googleapis_com_data_access`
+),
+/*
+ * RoutineDeletion: Routine deletion event
+ * https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/BigQueryAuditMetadata#routinedeletion
+ */
+routineDeletionEvent AS (
+  SELECT
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineDeletion.routine.routineName') AS routineDeletionName,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineDeletion.routine.createTime') AS routineDeletionCreateTime,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineDeletion.routine.updateTime') AS routineDeletionUpdateTime,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineDeletion.reason') AS routineDeletionReason,
+    JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+      '$.routineDeletion.jobName') AS routineDeletionJobName,
+    CONCAT(
+      SPLIT(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+            '$.routineDeletion.jobName'),
+            "/")[SAFE_OFFSET(1)],
+      ":",
+      SPLIT(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.routineDeletion.jobName'),
+            "/")[SAFE_OFFSET(3)]
+    ) AS jobId,
+  FROM `project_id.dataset_id.cloudaudit_googleapis_com_data_access`
+) -- End of WITH clauses
+SELECT
+  principalEmail AS principalEmail,
+  callerIp,
+  serviceName,
+  methodName,
+  eventName,
+  projectId,
+  jobId,
+  IF(jobChangeJobName IS NULL, False, True) AS hasJobChangeEvent,
```

Comment:
Are there any cases where multiple of these are true? If not, I'd change the naming to `is` as opposed to `has`

Reply:
Yes there are cases of multiple being true